src/interface/src/app/treatments/treatment-area-layer/treatment-area-layer.component.ts

```diff
@@ -0,0 +1,52 @@
+import { Component, inject, Input, OnInit } from '@angular/core';
+import { ActivatedRoute } from '@angular/router';
+import {
+  FeatureComponent,
+  GeoJSONSourceComponent,
+  LayerComponent,
+} from '@maplibre/ngx-maplibre-gl';
+import { PlanService } from '@services';
+import { Geometry } from 'geojson';
+import { GeoJSONSource, Map as MapLibreMap } from 'maplibre-gl';
+
+@Component({
+  selector: 'app-treatment-area-layer',
+  standalone: true,
+  imports: [FeatureComponent, GeoJSONSourceComponent, LayerComponent],
+  templateUrl: './treatment-area-layer.component.html',
+  styleUrl: './treatment-area-layer.component.scss',
+})
+export class TreatmentAreaLayerComponent implements OnInit {
+  @Input() mapLibreMap!: MapLibreMap;
+
+  planService: PlanService = inject(PlanService);
+  route: ActivatedRoute = inject(ActivatedRoute);
+
+  planId!: string;
+
+  readonly sourceName = 'tratment-planing-area';
+
+  polygonGeometry: Geometry = {
+    type: 'Polygon',
+    coordinates: [[]],
+  };
+
+  ngOnInit(): void {
+    this.route.paramMap.subscribe((params) => {
+      this.planId = params.get('planId') || '';
```

Comment:
Question, maybe the `planId` can be an @Input instead? so we don't couple routing stuff on this component, and keep it all on the parent one? What do you think? Might simplify this one a bit.
Another alternative, we are already setting some of the route info in TreatmentsState, maybe we should add the planId there? Like maybe update TreatmentsState.setInitialState to also store the planId ?

Reply:
That is a good idea, let me take a look <3